sdk/azcore/runtime/request.go

```diff
@@ -94,7 +95,9 @@ func MarshalAsByteArray(req *policy.Request, v []byte, format Base64Encoding) er
 
 // MarshalAsJSON calls json.Marshal() to get the JSON encoding of v then calls SetBody.
 func MarshalAsJSON(req *policy.Request, v interface{}) error {
-	v = cloneWithoutReadOnlyFields(v)
+	if omit := os.Getenv("AZURE_SDK_GO_OMIT_READONLY"); omit != "" {
```

Comment:
`AZURE_SDK_GO_OMIT_READONLY=false` gets the same behavior as `AZURE_SDK_GO_OMIT_READONLY=true`. I'm okay with that provided we document that any value toggles the behavior, but then I think we should allow "":
```suggestion
	if _, ok := os.LookupEnv("AZURE_SDK_GO_OMIT_READONLY"); ok {
```

Reply:
The nuance here that I probably should have included in the PR description is that we don't want to publicly document this env var. The thinking is as follows.
- if a service is rejecting a request due to mutation of a RO field, the customer would be required to update their code
- if a service is rejecting a request due to inclusion of a RO field that hasn't been mutated, then the service must be fixed per the ARM guidelines. customers can use this escape hatch to get unblocked in the meantime

Ideally, support for this env var would be removed at some point, and the fact that it's undocumented _should_ in theory give us the latitude to do so (similar to undocumented compiler switches).

The undocumented aspect is why I was a bit loose with the allowed values, but maybe that wasn't such a great idea. I'll clean that up.

Reply:
I've changed this to only accept `true` to enable this.